app/services/file_upload/file_uploader.rb

```diff
@@ -0,0 +1,15 @@
+module FileUpload
```

Comment:
feel like this file might be noise.. in that it could easily be abstracted as a method on the Base class removing the need for one additional layer of indirection..

However I don't feel super strongly about this...

Reply:
Used a factory because the underlying uploader can be whatever and the code itself doesnt care.